src/core/Events.js

```diff
@@ -216,19 +216,47 @@ export var Events = {
 	},
 
 	// @method listens(type: String, propagate?: Boolean): Boolean
+	// @method listens(type: String, fn: Function, context?: Object, propagate?: Boolean): Boolean
 	// Returns `true` if a particular event type has any listeners attached to it.
 	// The verification can optionally be propagated, it will return `true` if parents have the listener attached to it.
-	listens: function (type, propagate) {
+	listens: function (type, fn, context, propagate) {
 		if (typeof type !== 'string') {
 			console.warn('"string" type argument expected');
 		}
+
+		if (typeof fn === 'boolean') {
+			propagate = fn;
+			fn = undefined;
+			context = undefined;
+		}
+
 		var listeners = this._events && this._events[type];
-		if (listeners && listeners.length) { return true; }
+		if (listeners && listeners.length) {
+			if (fn) {
+				// we don't want to overwrite the context variable, because it is propagated
+				var ctx = context;
+				if (this === ctx) {
+					ctx = undefined;
+				}
+				// find fn
+				for (var i = 0, len = listeners.length; i < len; i++) {
+					var l = listeners[i];
+					if (l.ctx !== ctx) {
+						continue;
+					}
+					if (l.fn === fn) {
+						return true;
+					}
+				}

```

Comment:
Nit: we can write this in a more compact way for clarity:

```suggestion
				var ctx = context === this ? undefined : context;
				for (var i = 0, len = listeners.length; i < len; i++) {
					if (listeners[i].ctx === ctx && listeners[i].fn === fn) {
						return true;
					}
				}
```

Reply:
I copied it from:
https://github.com/Leaflet/Leaflet/blob/118ca443509476beec7f99e6d24694e4b9c1a6f2/src/core/Events.js#L164-L179

I suggest to use following from `_on` (same as your suggestion) and then use the new `listens` in `_on`
https://github.com/Leaflet/Leaflet/blob/118ca443509476beec7f99e6d24694e4b9c1a6f2/src/core/Events.js#L120-L125

Reply:
Sounds good to me.